kubernetes/utils/deserialize.py

```diff
@@ -0,0 +1,285 @@
+        Default is False.
+
+    Raises:
+        FailToLoadError which holds list of load failures..
+    """
+    with open(path.abspath(json_file)) as file:
+        data = file.read()
+        # python 2 has problems decoding unicode from json.load
+        # use yaml.load instead
+        if sys.version_info[0] < 3:
+            data = yaml.safe_load(data)
+        else:
+            data = json.loads(data)
+        obj = load_from_dict(data, klass=klass, verbose=verbose)
+
+        return obj
+
+
+def load_from_yaml(yaml_file, klass=None, verbose=False):
+    """
+    Perform an action from a yaml file. Pass True for verbose to
+    print additional information.
+    Input:
+    yaml_file: string. Contains the path to yaml file.
+    klass: class literal for deserialized object,
+    or string of class name.
+    verbose: If True, print information about the object lookup info.
+        Default is False.
+
+    Raises:
+        FailToLoadError which holds list of load failures.
+    """
+    with open(path.abspath(yaml_file)) as f:
+        yml_document_all = yaml.safe_load_all(f)
+        failures = []
+        objs = []
+        for yml_document in yml_document_all:
+            try:
+                obj = load_from_dict(
+                    yml_document, klass=klass, verbose=verbose)
+                # Prevent returning list of lists when doing multi
+                # document yaml
+                if isinstance(obj, list):
+                    objs.extend(obj)
+                else:
+                    objs.append(obj)
+
+            except FailToLoadError as failure:
+                failures.extend(failure.load_exceptions)
+
+        if objs:
+            result = objs if len(objs) > 1 else objs[0]
+
+        if failures:
+            raise FailToLoadError(failures)
+
+        return result
+
+
+def response_type_from_dict(data, verbose=False):
+    """
+    Helper function to do source code lookup and extract the
+    response_type of an kubernetes object.
+    Input:
+    data: a dictionary holding a valid kubernetes object
+    verbose: If True, print confirmation from the create action.
+        Default is False.
+
+    Raises:
+        FailToLoadError which holds list of load failures.
+    """
+    response_type = None
+    failures = []
+
+    group, _, version = data["apiVersion"].partition("/")
+    if version == "":
+        version = group
+        group = "core"
+    # Take care for the case e.g. api_type is "apiextensions.k8s.io"
+    # Only replace the last instance
+    group = "".join(group.rsplit(".k8s.io", 1))
+    # convert group name from DNS subdomain format to
+    # python class name convention
+    group = "".join(word.capitalize() for word in group.split("."))
+    fcn_to_call = "{0}{1}Api".format(group, version.capitalize())
+    k8s_api = getattr(client, fcn_to_call)
+    # Replace CamelCased action_type into snake_case
+    kind = data["kind"]
```

Comment:
The dict can be either a list of objects or a object, e.g. the following two types of responses
```
$ kubectl get po -ojson
{
    "apiVersion": "v1",
    "items": [
        {
            "apiVersion": "v1",
            "kind": "Pod",
            "metadata": {
...
```
v.s.
```
$ kubectl get po nginx-deployment-54f57cf6bf-cs6t7 -ojson
{
    "apiVersion": "v1",
    "kind": "Pod",
    "metadata": {
...
```
in the first case, the lookup function should be `list_namespaced_pod_with_http_info`

Reply:
The implementation has been changed to return k8s list objects as requested.